localstack/utils/http.py

```diff
@@ -60,7 +60,7 @@ def canonicalize_headers(headers: Union[Dict, CaseInsensitiveDict]) -> Dict:
     def _normalize(name):
         if name.lower().startswith(ACCEPT):
             return name.lower()
-        return name
+        return name.title()
```

Comment:
This is not per design AWS behavior. For example, when using apigateway with a lambda proxy integration to a lambda, the event payload contains the casing as sent (see the `test-header`):
```
        "headers": {
          "Accept": "*/*",
          "Accept-Encoding": "gzip, deflate",
          "CloudFront-Forwarded-Proto": "https",
          "CloudFront-Is-Desktop-Viewer": "true",
          "CloudFront-Is-Mobile-Viewer": "false",
          "CloudFront-Is-SmartTV-Viewer": "false",
          "CloudFront-Is-Tablet-Viewer": "false",
          "CloudFront-Viewer-ASN": "<cloudfront-asn:1>",
          "CloudFront-Viewer-Country": "<cloudfront-country:1>",
          "Host": "<id:1>.execute-api.<region>.amazonaws.com",
          "User-Agent": "python-requests/testing",
          "Via": "<via:1>",
          "X-Amz-Cf-Id": "<cf-id:1>",
          "X-Amzn-Trace-Id": "<trace-id:1>",
          "X-Forwarded-For": "<source-ip:1>, <ip>",
          "X-Forwarded-Port": "443",
          "X-Forwarded-Proto": "https",
          "tEsT-HEADeR": "aValUE"
        },
```